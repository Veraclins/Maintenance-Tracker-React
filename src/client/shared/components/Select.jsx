
import React from 'react';
import { PropTypes } from 'prop-types';

const SelectInput = ({
  name,
  handleChange,
  placeholder,
  inputLabel,
  required,
  value,
  options,
}) => (
  <React.Fragment>
    <div className="field hs-form-field">
      <label className="hs-form-label" htmlFor={name}>
        <span>{inputLabel}{' '}</span>
        {required ? <span className="hs-form-required">*</span> : false}
      </label>
      <div className="input">
        <select
          className="hs-input"
          name={name}
          value={value}
          onChange={handleChange}
          options={options}
          placeholder={placeholder}
        >
          {options.map(option => (
            <option value={option.value} key={option.label}>{option.label}</option>
          ))}
        </select>
      </div>
    </div>
  </React.Fragment>
);

SelectInput.propTypes = {
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  required: PropTypes.bool.isRequired,
  inputLabel: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default SelectInput;
