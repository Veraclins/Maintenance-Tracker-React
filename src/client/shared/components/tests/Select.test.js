import React from 'react';
import { shallow } from 'enzyme';

import { SelectInput } from '../Select';

const mockFunction = jest.fn();
describe('Test the Form component', () => {
  it('renders without crashing', () => {
    const options = [
      { value: '', label: 'Select Device Type' },
      { value: 'Laptop', label: 'Laptop Pc' },
    ];
    const wrapper = shallow(<SelectInput
      name="input"
      required
      options={options}
      inputLabel="Text input"
      value=""
      placeholder="enter text"
      autoComplete="on"
      handleChange={mockFunction}
    />);
    expect(wrapper.find('select')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
