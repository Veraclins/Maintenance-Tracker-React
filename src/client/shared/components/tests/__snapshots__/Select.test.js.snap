// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Form component renders without crashing 1`] = `
<Fragment>
  <div
    className="field hs-form-field"
  >
    <label
      className="hs-form-label"
      htmlFor="input"
    >
      <span>
        Text input
         
      </span>
      <span
        className="hs-form-required"
      >
        *
      </span>
    </label>
    <div
      className="input"
    >
      <select
        className="hs-input"
        name="input"
        onChange={[MockFunction]}
        options={
          Array [
            Object {
              "label": "Select Device Type",
              "value": "",
            },
            Object {
              "label": "Laptop Pc",
              "value": "Laptop",
            },
          ]
        }
        placeholder="enter text"
        value=""
      >
        <option
          key="Select Device Type"
          value=""
        >
          Select Device Type
        </option>
        <option
          key="Laptop Pc"
          value="Laptop"
        >
          Laptop Pc
        </option>
      </select>
    </div>
  </div>
</Fragment>
`;
